Apply config.c refactorings

clumens opened the pull request:
Continuing with #82, this is the next couple patches that address the use of global variables in config.[ch].  I made a couple changes to the original patches - get rid of `_ptr` from the variable names, add braces around conditional blocks in code that this touches, and removing a function instead of marking it as unused.

I tested this only briefly, by installing updated packages and making sure that booth started and could grant tickets.

Note - this builds on top of #147 so that needs to be merged first, then I'll rebase this one and it can be merged as well.

clumens commented:
Rebased on top of just merged mock patches.

jfriesse commented:
I've checked commits twice and it looks there is no obvious problem codewise, but because you are native english speaker you might check comments/commit msg. I'm not native speaker so it is hard for me, but honestly never heard of "superjacent" (for example) and "Currently it means checking that login user/group indeed exists," also looks suspicious (I would personally use "really" instead of "indeed").

Anyway, I you think comments/commit msg is ok I will just give it formal ack right then.

clumens commented:
> I've checked commits twice and it looks there is no obvious problem codewise, but because you are native English speaker you might check comments/commit msg. I'm not native speaker so it is hard for me, but honestly never heard word "superjacent" (for example) and "Currently it means checking that login user/group indeed exists," also looks suspicious (I would personally use "really" instead of "indeed"). It just doesn't sound for me like native English.

I had to look it up and "superjacent" is a word.  I've never heard it either.

I was trying to keep the commit messages the same to stay as close to the original commit (and original authorship) as possible, but perhaps I should go ahead and reword them to be a little easier to understand.  I'm not sure if I should just make new commits with me as the author and somehow credit poki in the commit message, or what.

clumens commented:
Updated with new commit messages and comment for check_config, changed the author to myself (I've made various changes in this PR, I hope this is okay), and added poki as Co-authored-by.  If this looks good, this is the process I'll use going forward.  This is ready for ACK/NAK.

jfriesse reviewed (approved):
Perfect, thanks, now it looks like English and much easier to understand. I think Co-authored is good idea so ACK.
